### PR TITLE
Added rand.h implementation

### PR DESCRIPTION
**Pull request by NouemanKHAL**
Supported types for now:  
- Numbers: Int, Long, Long Long, Double
- Chars: Char, Lower, Upper, Alpha, Digit
- String

### CODE REVIEW

**Review by Eroui — APPROVED** [review]
The  rand.h content is 4 spaces tab, I would go for 2 tabs. Otherwise LGTM

File: templates/resolver.go [hunk 1]
```diff
@@ -22,9 +22,90 @@ int main() {
...
+	UPPERCASE,
+	MIXEDCASE
+};
 class Rand {
+private:
+	int seed;
+	void init() {
+		srand(seed);
+	}
+	void readArgs(int argc, char * argv[]) {
```
Comment (Eroui) [inline templates/resolver.go:39]: InitSeedFromArgs (Or something like this) is a better name than readArgs which might be a little bit confusing

File: templates/resolver.go [hunk 2]
```diff
@@ -22,9 +22,90 @@ int main() {
...
+		return Upper('a', 'a' + k % 26);
+	}
+	char Digit(char from = '0', char to = '9') {
+		return Char(from, to);
+	}
+	char Char(char from, char to) {
+		return static_cast <char> (Int(from, to));
+	}
+	// Strings
+	std::string String(size_t size, CASE mode = MIXEDCASE) {
```
Comment (Eroui) [inline templates/resolver.go:87]: What about a alpha numerical strings? I don't think this actually support that?

**Review by Eroui — APPROVED** [review]
Few other nit comments.

File: templates/resolver.go [hunk 3]
```diff
@@ -22,9 +23,115 @@ int main() {
...
+			seed = std::stoi(argv[1]);
+		}
+		else {
+			seed = 0;
+		}
+	}
+
 public:
-	Rand(int argc, char** argv) {
+	Rand(unsigned int _seed) : seed(_seed) {
```
Comment (chermehdi) [inline templates/resolver.go:53]: Why do we have two init methods ?

File: templates/resolver.go [hunk 4]
```diff
@@ -22,9 +23,115 @@ int main() {
...
+	// Numbers
+	int Int(int from, int to) {
+		return rand() % (to - from) + from;
+	}
+
+	long Long(long from, long to) {
+		return rand() % (to - from) + from;
+	}
+
+	long long LongLong(long long from, long long to) {
```
Comment (chermehdi) [inline templates/resolver.go:72]: either call it `Long` or call it `Int64`

File: templates/resolver.go [hunk 5]
```diff
@@ -22,9 +23,115 @@ int main() {
...
+		return from + tmp * (to - from);
+	}
+
+
+	// Chars
+	char Char(char from, char to) {
+		return static_cast <char> (Int(from, to));
+	}
+
+	char Lower(char from = 'a', char to = 'z') {
```
Comment (chermehdi) [inline templates/resolver.go:87]: Lower shouldn't expect parameters (it's implied in the name)

File: templates/resolver.go [hunk 6]
```diff
@@ -22,9 +23,115 @@ int main() {
...
+		return Char(from, to);
+	}
+
+	char Upper(char from = 'A', char to = 'Z') {
```
Comment (chermehdi) [inline templates/resolver.go:91]: The same as above.

File: templates/resolver.go [hunk 7]
```diff
@@ -22,9 +23,115 @@ int main() {
...
+		return Upper('a', 'a' + k % 26);
+	}
+
+	char Digit(char from = '0', char to = '9') {
+		return Char(from, to);
+	}
+
+
+	// Strings
+	std::string String(size_t size, CASE mode = MIXEDCASE) {
```
Comment (chermehdi) [inline templates/resolver.go:107]: When we talked about this offline I imagined having something like this:

```cpp
const int LOWER = 1
const int UPPER = 1 << 1
const int DIGITS = 1 << 2
// caller would do something like this.
rand.RandString(10, LOWER | UPPER)
```

**Review by chermehdi — CHANGES_REQUESTED** [review]
Generals comments:

- The indentation should be fixed to 2 spaces
- The class needs some docs on public methods, even if it looks obvious (with specified explicit bounds)

**Review by chermehdi — APPROVED** [review]

**Pull request merged**